subsys/bluetooth/services/ddfs.c

```diff
@@ -219,8 +221,10 @@ static ssize_t dm_write_ctrl_point(struct bt_conn *conn, const struct bt_gatt_at
 		if (ddfs_inst.cb && ddfs_inst.cb->dm_config_read) {
 			if (!ddfs_inst.cb->dm_config_read(&dm_config)) {
 				status = DDF_CP_RSP_SUCCESS;
-				ctrl_point_ind(conn, req->op, status, (uint8_t *)(&dm_config.mode),
-					       sizeof(uint8_t));
+				config_arr[0] = dm_config.mode;
+				config_arr[1] = dm_config.high_precision ? 1 : 0;
```

Comment:
Please hide the magic numbers behind proper macros (DM_HIGH_PRECISION_ENABLED/DISABLED)